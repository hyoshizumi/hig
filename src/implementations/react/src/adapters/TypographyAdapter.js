import * as HIG from 'hig-vanilla';
import * as PropTypes from 'prop-types';

import HIGElement from '../elements/HIGElement';
import createComponent from './createComponent';

class TypographyAdapter extends HIGElement {
  constructor(initialProps) {
    super(HIG.Typography, initialProps)
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];
      switch (propKey) {
        case 'type': {
          this.hig.setType(propValue);
          break;
        }
        case 'text': {
          this.hig.setText(propValue);
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const TypographyAdapterComponent = createComponent(TypographyAdapter);

TypographyAdapterComponent.propTypes = {
  type: PropTypes.string,
  text: PropTypes.string
};

TypographyAdapterComponent.__docgenInfo = {
  props: {
    type:    {
      description: "{String - 'stack', 'inline'} type of the typography"
    }, text: {
      description: "{String - styled or unstyled text to show inside the typography component"
    }
  }
};

export default TypographyAdapterComponent;