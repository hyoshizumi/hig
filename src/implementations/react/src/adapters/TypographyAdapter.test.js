import {mount} from 'enzyme';
import * as HIG from 'hig-vanilla';
import React from 'react';

import {default as Typography} from './TypographyAdapter';

const Context = props => {
  return (
    <Typography type={props.type} text={props.text}/>
  );
};

function setupReactContext(props) {
  const reactContainer = document.createElement("div");
  const reactWrapper = mount(<Context {...props}/>, {
                                                      attachTo: reactContainer
                                                    });
  return {reactWrapper, reactContainer}
}

function createHigContext(props) {
  const higContainer = document.createElement('div');

  const higSpacer = new HIG.Typography({...props});
  higSpacer.mount(higContainer);

  return {higContainer, higItem: higSpacer}
}

describe("Typography Adapter", () => {
  it('renders a basic typography adapter', () => {
    const props = {type: 'h1', text: "Come at me, bro!"}
    const {reactWrapper, reactContainer} = setupReactContext(props);
    const {higContainer, higItem} = createHigContext(props);

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstElementChild.outerHTML).toEqual(higContainer.firstElementChild.outerHTML);
  });

  describe('can update props', () => {
    const props = {type: 'h1', text: "Come at me, bro!"};

    const { higContainer, higItem } = createHigContext(props);
    const {reactWrapper, reactContainer} = setupReactContext(props);

    const newProps = {type: 'h3', text: 'Calm down, bro!'};


    //Update hig-vanilla instance
    higItem.setType(newProps.type);
    higItem.setText(newProps.text);

    //Update React Instance
    reactWrapper.setProps(newProps);

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    //Check them against each other
    expect(reactContainer.firstChild.outerHTML).toEqual(
      higContainer.firstChild.outerHTML
    );

  });
});