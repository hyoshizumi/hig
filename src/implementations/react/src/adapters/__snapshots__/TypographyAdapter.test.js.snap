// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"<span class=\\"hig__typography hig__typography__h3\\">Calm down, bro!</span>"`;

exports[`Typography Adapter renders a basic typography adapter 1`] = `"<span class=\\"hig__typography hig__typography__h1\\">Come at me, bro!</span>"`;
